project-base/app/tests/FrontendApiBundle/Functional/Cart/MergeCartsTest.php

```diff
@@ -96,6 +96,64 @@ public function testCartIsMergedAfterLogin(): void
         self::assertNull($oldCart);
     }
 
+    public function testCartIsOverwrittenAfterLogin(): void
+    {
+        /** @var \App\Model\Product\Product $anonymouslyAddedProduct */
+        $anonymouslyAddedProduct = $this->getReference(ProductDataFixture::PRODUCT_PREFIX . '5');
+        $anonymouslyAddedProductQuantity = 6;
+        $this->addProductToCustomerCart($anonymouslyAddedProduct, $anonymouslyAddedProductQuantity);
+
+        /** @var \App\Model\Product\Product $anonymouslyAddedProduct2 */
+        $anonymouslyAddedProduct2 = $this->getReference(ProductDataFixture::PRODUCT_PREFIX . '1');
+        $anonymouslyAddedProductQuantity2 = 1;
+        $this->addProductToCustomerCart($anonymouslyAddedProduct2, $anonymouslyAddedProductQuantity2);
+
+        $testCartUuid = CartDataFixture::CART_UUID;
+
+        $loginMutationWithCartUuid = 'mutation {
```

Comment:
We now use a better approach - the mutation should be put in a separate graphql file, see e.g. https://github.com/shopsys/shopsys/blob/14.0/project-base/app/tests/FrontendApiBundle/Functional/Product/ProductList/ProductListLoggedCustomerTest.php#L135

Reply:
I knew about that approach, but actually just copied the test "above" 😆 I will fix it.

Reply:
I rewrote the test "completely". Let's see if that's better 😆

Reply:
yeah, but the assertions are still weird, lets take a look at it together

Reply:
yeah, but the assertions are still weird, lets take a look at it together

Reply:
After discussing it on a call, we decided to leave as-is and disambiguate this at a later time.